runtime/lib/registry/Registry.cpp

```diff
@@ -15,8 +33,66 @@ namespace py = pybind11;
 // https://pybind11.readthedocs.io/en/stable/advanced/misc.html#common-sources-of-global-interpreter-lock-errors
 std::unordered_map<int64_t, py::function> *references;
 
+std::string libmlirpath;
+
+void convertResult(py::handle tuple)
+{
+
+    py::object unrankedMemrefPtrSizeTuple = tuple.attr("__getitem__")(0);
+
+    py::object unranked_memref = unrankedMemrefPtrSizeTuple.attr("__getitem__")(0);
+    py::object element_size = unrankedMemrefPtrSizeTuple.attr("__getitem__")(1);
+    py::object unranked_memref_ptr_int = unranked_memref.attr("value");
+
+    void *unranked_memref_ptr = (void *)(py::cast<long>(unranked_memref_ptr_int));
+    long e_size = py::cast<long>(element_size);
+
+    std::string libpath = libmlirpath + "/libmlir_c_runner_utils.so";
+    void *handle = dlopen(libpath.c_str(), RTLD_LAZY);
+    if (!handle) {
+        throw py::value_error(dlerror());
```

Comment:
If you throw here with the above file open, it can cause issues. Let's make the RAII like class for open/close files if exceptions are to be thrown and handled by the Python layer.

Reply:
And here: https://github.com/PennyLaneAI/catalyst/pull/610/commits/c2c4134926e02f336d028b71021b7b13599cc26a